subsys/shell/shell_rtt.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2018 Makaio GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <shell/shell_rtt.h>
+#include <init.h>
+#include <rtt/SEGGER_RTT.h>
+
+SHELL_RTT_DEFINE(shell_transport_rtt);
+SHELL_DEFINE(rtt_shell, "rtt:~$ ", &shell_transport_rtt, 10,
+	     SHELL_FLAG_OLF_CRLF);
+
+
+static struct k_thread rtt_rx_thread;
+static K_THREAD_STACK_DEFINE(rtt_rx_stack, 1024);
+
+static void shell_rtt_rx_process(struct shell_rtt *sh_rtt)
+{
+	unsigned int count;
+#ifdef SHELL_BACKEND_RTT_IRQLOCK
+	unsigned int key;
+#endif
+
+	while (true)
+	{
+#ifdef SHELL_BACKEND_RTT_IRQLOCK
+		key = irq_lock();
+#else
+		SEGGER_RTT_LOCK();
+#endif
+		count = SEGGER_RTT_ReadNoLock(0, sh_rtt->rx, sizeof(sh_rtt->rx));
```

Comment:
Why not use just `SEGGER_RTT_Read`. It internally calls Segger_RTT_Lock which currently by default is asm implementation of irq_lock and once @pkral78 PR goes in will become mutex. Imo, this option is redundant.

Reply:
@nordic-krch have right. Just note that reading use downBuffer, logger only write to upBuffer, so these two do not clash. But SEGGER API do not allows use two mutexes, I comment that in my PR.

Reply:
@nordic-krch so I'll remove both lock logics and just call SEGGER_RTT_Read?

Reply:
@nordic-krch @pkral78 any advice?

Reply:
@derchrismakaio Juste replace SEGGER_RTT_ReadNoLock with SEGGER_RTT_Read and remove LOCK/UNLOCK macros that should be enough.